### PR TITLE
Add support for multiple grayskull clusters at once on AWS

### PR DESCRIPTION
**Pull request by aperullo**
This PR adds the ability to deploy grayskull multiple times on AWS. All in seperate VPCs and with seperate security groups. This prevents the previous conflicts of shared resources, which kept us from running more than one grayskull instance at a time.

### CODE REVIEW

File: kubernetes/ansible/playbooks/grayskull.yml [hunk 1]
```diff
@@ -81,7 +81,7 @@
       mode: '0755'
 
   - name: Customization Files | Move to kube-master
-    copy:
+    template:
```
Comment (njbarber) [inline kubernetes/ansible/playbooks/grayskull.yml]: Why was this changed? Is this just another way of copying?

File: kubernetes/terraform/k8s_cluster/internet_gateway.tf [hunk 2]
```diff
@@ -0,0 +1,27 @@
+resource "aws_internet_gateway" "k8s_igw" {
+  vpc_id = "${aws_vpc.k8s_vpc.id}"
+
+  tags = {
+    Name = "k8s_${terraform.workspace}_igw"
```
Comment (njbarber) [inline kubernetes/terraform/k8s_cluster/internet_gateway.tf]: Where does `terraform.workspace` get interpolated from? The workspace where any `terraform` command is run?

File: kubernetes/terraform/main.tf [hunk 3]
```diff
@@ -1,4 +1,5 @@
 module "k8s_cluster" {
   source = "./k8s_cluster"
   external_ip_list = var.external_ip_list
```
Comment (njbarber) [inline kubernetes/terraform/main.tf]: You have `external_ip_list` and `public_dns` defined `variables.tf` in this directory, and also in `variables.tf` in the `k8s_cluster` subdirectory. What's the purpose of having it in both places? Are you planning on having more modules referenced by `source` here and effectively creating/overwriting defaults?

File: kubernetes/terraform/k8s_cluster/subnet.tf [hunk 4]
```diff
@@ -12,6 +14,8 @@ resource "aws_subnet" "k8s_subnet_2" {
   vpc_id     = "${aws_vpc.k8s_vpc.id}"
```
Comment (njbarber) [inline kubernetes/terraform/k8s_cluster/subnet.tf]: What is the purpose of this subnet? I don't see it referenced anywhere else.

File: kubernetes/terraform/k8s_cluster/subnet.tf [hunk 5]
```diff
@@ -22,6 +26,8 @@ resource "aws_subnet" "k8s_subnet_3" {
...
```
Comment (njbarber) [inline kubernetes/terraform/k8s_cluster/subnet.tf]: Same as above.

File: kubernetes/terraform/k8s_cluster/internet_gateway.tf (hunk 2 above)
Comment (njbarber) [reply kubernetes/terraform/k8s_cluster/internet_gateway.tf]: Not related to this PR, but there are other files in this directory that have `Environment = terraform.workspace`. You may want to check that these work the same and pick one form to be consistent.

File: kubernetes/terraform/k8s_cluster/internet_gateway.tf [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    Environment = "${terraform.workspace}"
+  }
+}
+
+resource "aws_route_table" "k8s_rt" {
```
Comment (njbarber) [inline kubernetes/terraform/k8s_cluster/internet_gateway.tf]: Do you need to link each `aws_subnet` to a `aws_route_table` through a `aws_route_table_association`? The example I saw is:

https://github.com/nickcharlton/terraform-aws-vpc/blob/master/vpc.tf

Comment (aperullo) [reply kubernetes/terraform/k8s_cluster/internet_gateway.tf]: It seems to work without doing that, so I'm inclined to say It isn't necessary. What does binding to each subnet do?

Comment (oconnormi) [reply kubernetes/terraform/k8s_cluster/internet_gateway.tf]: only if you need to have multiple route tables in the vpc where some of them were specific to a certain set of subnets. We actually should probably be using [default route table](https://www.terraform.io/docs/providers/aws/r/default_route_table.html)

File: kubernetes/ansible/playbooks/grayskull.yml (hunk 1 above)
Comment (oconnormi) [reply kubernetes/ansible/playbooks/grayskull.yml]: Template renders the file instead of just copying it

File: kubernetes/terraform/k8s_cluster/internet_gateway.tf (hunk 2 above)
Comment (oconnormi) [reply kubernetes/terraform/k8s_cluster/internet_gateway.tf]: The Environment one is actually a tag which is really just metadata used to help filter out stuff when searching for things in the aws console.

File: kubernetes/terraform/main.tf (hunk 3 above)
Comment (oconnormi) [reply kubernetes/terraform/main.tf]: Its required to have them in both places. The lower level one defines variables that can be passed into the k8s_cluster module as inputs (helps make it reusable) and the one thats above that defines variables that can be passed into the main module. Right now we are using the same variable names which is what makes it look slightly confusing.

File: kubernetes/terraform/k8s_cluster/subnet.tf (hunk 4 above)
Comment (oconnormi) [reply kubernetes/terraform/k8s_cluster/subnet.tf]: We'll eventually be using it once we spread the load across the subnets. The subnets should each be in a different availability zone, so when we spread the load across multiple availability zones (for resiliency) we'll need the other subnets

File: kubernetes/ansible/playbooks/grayskull.yml (hunk 1 above)
Comment (njbarber) [reply kubernetes/ansible/playbooks/grayskull.yml]: Yep. I didn't see the commit where templating was added to those files...

File: kubernetes/terraform/k8s_cluster/internet_gateway.tf (hunk 6 above)
Comment (aperullo) [reply kubernetes/terraform/k8s_cluster/internet_gateway.tf]: Oh this definitely looks preferable. WIll do

File: kubernetes/terraform/k8s_cluster/internet_gateway.tf (hunk 2 above)
Comment (aperullo) [reply kubernetes/terraform/k8s_cluster/internet_gateway.tf]: picked `terraform.environment` except where necessary for string interpolaton
